### PR TITLE
Adding babel preinstall script to support node 4.5

### PR DESCRIPTION
**Pull request by chfritz**
This PR adds a preinstall script to package.json that transpiles the files with babel's es2015 preset. This adds support for Node 4.5. I will create a similar PR for rosnodejs.

### CODE REVIEW

**Comment by chris-smith** [comment]
Glad this is happening - I have been meaning to do something about this for a while but I don't have a ton of experience with best practices for this sort of thing... What I have [read](http://stackoverflow.com/questions/29738381/how-to-publish-a-module-written-in-es6-to-npm) usually advocates using babel as a `prepublish` step though rather than a `pre/post-install` step. I think, depending on the setup people have, a `pre-` or `postinstall` step could fail because of `Node`/`babel` version issues.

Removing it as an `-install` step would require people developing for `rosnodejs` to be on a newer version of `Node` though - or to deal with transpiling it themselves when developing.

**Comment by IanTheEngineer** [comment]
+1 for the `prepublish` step. In my mind it makes sense to require `rosnodejs` dev's to be on the most current LTS of Node. It is easier to manage, and keeps the code more consistent.

**Comment by chfritz** [comment]
I agree that `prepublish` is cleaner, but it also means that people won't be able to use github based dependencies (like the one currently used in rosnodejs for RethinkRobotics-opensource/ros_msg_utils), only versions actually published to the npm registry -- I _think_.

Babel is a devDependency, so it will be installed in `node_modules/ros_msg_utils/node_modules/.bin/babel` and according to the documentation, that is the path being used in script, though I haven't confirmed that.

As a middle ground, we could create a `compile` script, and set prepublish to `npm run compile`. That won't fully resolve the above issue regarding github dependencies, but makes it easier to manually handle that -- devs would need to run `npm install` followed by something like `cd node_modules/rodnodejs && npm run compile`.

**Comment by chris-smith** [comment]
The `compile` script sounds good to me. Github based dependencies are an issue with the `prepublish` step that I haven't seen a great answer to - workarounds, yes, but...

Publishing both packages again is something I've meant to do though. Publishing `rosnodejs` should possibly wait until the new `gennodejs` is in kinetic but `ros_msg_utils` would be fine to go out on its own.

**Comment by chris-smith** [comment]
I'm still worried about the `preinstall` step - I think it should be run before anything else is done and therefore babel won't necessarily be installed when it gets run. There _is_ a [bug](https://github.com/npm/npm/issues/10379) in some versions of npm that execute `preinstall` out of order though - which should be fixed in `npm 3.10.8`. @IanTheEngineer had issues running with the `preinstall` script because he's using `node v6.9.0` which comes with the fixed version of `npm` - we had to install babel specifically before being able to install the package itself.

**Comment by chfritz** [comment]
Well, the way that I have it right now the `preinstall` step shouldn't do anything *unless* you are fetching the package through a non-published channel, i.e., not from npm but, e.g., from github. So in the usual use case nothing would happen. Does that address your concern, or is there still a case where this would break for you? -- btw, it shouldn't be necessary to install babel manually. The npm scripts use the binaries installed in the node_modules/.bin path itself. Is that no longer the case with the new version of npm in node 6.9?

**Comment by chfritz** [comment]
OK. I've found another solution for my problem that doesn't require the `preinstall` script. I've removed it from the two package definitions (here and in `rosnodejs`).

File: package.json [hunk 1]
```diff
@@ -17,7 +19,13 @@
     "url": "git://github.com/RethinkRobotics-opensource/ros_msg_utils"
   },
   "devDependencies": {
+    "babel-cli": "^6.18.0",
+    "babel-preset-es2015": "^6.18.0",
     "chai": "^3.5.0",
     "mocha": "^3.0.2"
+  },
+  "dependencies": {
```
Comment (chris-smith) [inline package.json]: does `babel` still need to be here in addition to `devDependencies`?

**Review by chris-smith — COMMENTED** [review]
Just the one question

**Comment by chfritz** [comment]
ah, yes. Good catch. Thanks.

**Review by chris-smith — APPROVED** [review]

**Pull request merged**